src/python/pants/notes/2.7.x.md

```diff
@@ -0,0 +1,37 @@
+## 2.7.0.dev0 (Jul 09, 2021)
+
+### New Features
+
+* Add skimage module mapping for dependency inference ([#12301](https://github.com/pantsbuild/pants/pull/12301))
+
+* add experimental terraform plugin with fmt and lint rules ([#11973](https://github.com/pantsbuild/pants/pull/11973))
+
+* Include experimental Go backends in pantsbuild.pants ([#12290](https://github.com/pantsbuild/pants/pull/12290))
+
+* Add support for MyPy report files ([#12271](https://github.com/pantsbuild/pants/pull/12271))
+
+* Don't tailor a python_library for a solitary __init__.py. ([#12288](https://github.com/pantsbuild/pants/pull/12288))
+
+* Support running tailor on subdirs. ([#12262](https://github.com/pantsbuild/pants/pull/12262))
+
+* Add JavacSubsystem and JavacBinary, allowing user-configurable JVM selection ([#12283](https://github.com/pantsbuild/pants/pull/12283))
+
+* Revert "Deprecate unused `--process-execution-local-enable-nailgun` (#11600)" ([#12257](https://github.com/pantsbuild/pants/pull/12257))
+
+* Apply a concurrency limit to remote store/cache/execution gRPC requests ([#12256](https://github.com/pantsbuild/pants/pull/12256))
```

Comment:
Technically a new feature...but I think probably better as User API change because it changes the default behavior for remote cache/execution, w/ a toggle to further change it.